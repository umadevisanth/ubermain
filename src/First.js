import React, { useState } from 'react';
import { View, Text, SafeAreaView, TouchableOpacity, TextInput } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import Globalstyle from './Globalstyle';
import { Feather } from '@expo/vector-icons';


const First = () => {
  const navigation = useNavigation();
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [firstNameError, setFirstNameError] = useState('');
  const [lastNameError, setLastNameError] = useState('');
  const first = {
    name: 'What`s Your Name?',
    next: 'Next'
  };

  const handleNextPress = () => {
    const minNameLength = 2; // Minimum length for name fields

    // Check if first name is empty or does not meet minimum length
    if (!firstName || firstName.length < minNameLength) {
      setFirstNameError('First name is required .');
      return;
    }

    // Check if last name is empty or does not meet minimum length
    if (!lastName || lastName.length < minNameLength) {
      setLastNameError('Last name is required .');
      return;
    }

    // Check if first name contains numeric characters
    if (/\d/.test(firstName)) {
      setFirstNameError('First name cannot contain numeric characters.');
      return;
    }

    // Check if last name contains numeric characters
    if (/\d/.test(lastName)) {
      setLastNameError('Last name cannot contain numeric characters.');
      return;
    }

    // Reset errors if validation passes
    setFirstNameError('');
    setLastNameError('');

    // Navigate to the next screen if validation passes
    navigation.navigate('Profile');
  };

  return (
    <SafeAreaView style={Globalstyle.firstsafeview}>
      <View style={Globalstyle.firstviewtext}>
        <TouchableOpacity onPress={() => navigation.goBack('Social')}>
          <Feather name="arrow-left" size={24} color="#EDF6FF" style={Globalstyle.First_left} />

        </TouchableOpacity>
      </View>
      <View style={{ flex: 5 }}>
        <View style={Globalstyle.firstviewtexttext}>
          <Text style={Globalstyle.First_name}>{first.name}</Text>
        </View>

        <View style={Globalstyle.First_view}>
          <View >
            <TextInput
              style={{ color: '#EDF6FF', width: '35%', fontSize: 18, borderBottomWidth: 1, borderColor: firstNameError ? 'red' : '#EDF6FF', padding: 20, width: 150 }}
              placeholder="First" placeholderTextColor={'#EDF6FF'}
              value={firstName}
              onChangeText={(text) => { setFirstName(text.replace(/[^A-Za-z]/g, '')); setFirstNameError(''); }}
            />
            {firstNameError ? <Text style={Globalstyle.firstError}>{firstNameError}</Text> : null}
          </View>
          <View>
            <TextInput
              style={{ color: '#EDF6FF', width: '35%', fontSize: 18, borderBottomWidth: 1, borderColor: lastNameError ? 'red' : '#EDF6FF', padding: 20, width: 150 }}
              placeholder="Last" placeholderTextColor={'#EDF6FF'}
              value={lastName}
              onChangeText={(text) => { setLastName(text.replace(/[^A-Za-z]/g, '')); setLastNameError(''); }}
            />
            {lastNameError ? <Text style={Globalstyle.lastError}>{lastNameError}</Text> : null}
          </View>
        </View>
      </View>
      <View style={{ flex: 1 }}>
        <TouchableOpacity
          style={Globalstyle.First_next}
          onPress={handleNextPress}
        >
          <Text style={Globalstyle.First_nexttext}>{first.next}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default First;