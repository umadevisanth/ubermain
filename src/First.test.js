

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import First from '../src/First';

// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
    AntDesign: 'AntDesign', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('First component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<First/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

