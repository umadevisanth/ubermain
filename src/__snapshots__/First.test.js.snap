// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <AntDesign
        color="black"
        name="left"
        size={20}
        style={
          {
            "color": "#979797",
            "height": 16,
            "left": 53,
            "top": 96,
            "width": 16,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 30,
          "height": 33,
          "left": 37,
          "top": 131,
          "width": 349,
        }
      }
    >
      What's Your Name?
    </Text>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": "auto",
        "marginRight": "auto",
        "top": 190,
        "width": "85%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="First *"
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#EDF6FF",
          "color": "#EDF6FF",
          "fontSize": 20,
          "padding": 20,
          "width": 150,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Last *"
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#EDF6FF",
          "color": "#EDF6FF",
          "fontSize": 20,
          "padding": 20,
          "width": 150,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#EDF6FF",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 58,
        "justifyContent": "center",
        "left": 37,
        "opacity": 1,
        "top": 500,
        "width": 341,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 21,
          "left": 150,
        }
      }
    >
      Next
    </Text>
  </View>
</RCTSafeAreaView>
`;
